modules/nextflow/src/main/groovy/nextflow/extension/ChannelExtensionDelegate.groovy

```diff
@@ -44,10 +43,12 @@ class ChannelExtensionDelegate implements DelegatingPlugin {
 
     private Session getSession() { Global.getSession() as Session }
 
-    final private Map<String, ChannelFactory> channelFactories = new HashMap<>()
-
     final private Map<String,Object> operatorExtensions = new HashMap<>()
 
+    final private Map<String,Tuple2<String,Object>> aliasOperatorExtensions = new HashMap<>()
+
+    final private Map<String,Tuple2<String,ChannelFactoryInstance>> aliasFactoryExtensions = new HashMap<>()
+
```

Comment:
Let's use a custom class instead `Tuple2` that should help a bit to make the underlying logic more accessible. Same for `aliasOperatorExtensions`